controller/repliesController.js

```diff
@@ -102,7 +102,67 @@ const createReply = async (req, res, next) => {
   }
 };
 
+const flagCommentReplies = async (req, res, next) => {
+  try {
+    //validation should be done via middleware
+    //ownerId in body also needs to be validated
+
+    const { commentId, replyId } = req.params;
+    const { ownerId } = req.body;
+
+    if (!ObjectId.isValid(commentId)) {
+      return next(new CustomError(422, " Invalid comment Id "));
+    }
+
+    if (!ObjectId.isValid(replyId)) {
+      return next(new CustomError(422, " Invalid reply Id "));
+    }
+    const reply = await Replies.findOne({
+      _id: replyId,
+      commentId: commentId,
+    });
+
+    if (!reply) {
+      next(
+        new CustomError(
+          404,
+          `Reply with the ID ${replyId} doesn't exist or has been deleted`
+        )
+      );
+      return;
+    }
+
+    //flag comment reply by pushing ownerId into flags array
+    if (!reply.flags.includes(ownerId)) {
+      reply.flags.push(ownerId);
+    }
```

Comment:
You need to add an else statement that will also remove the ownerId from the flags array if it already exists.

Think of it as a toggle.